modules/cef/event-format-parser.c

```diff
@@ -0,0 +1,323 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <string.h>
+#include "event-format-parser.h"
+#include "filterx-func-parse-cef.h"
+#include "filterx/object-string.h"
+#include "filterx/object-primitive.h"
+#include "filterx/expr-literal.h"
+#include "filterx/expr-literal-generator.h"
+#include "filterx/filterx-eval.h"
+#include "filterx/filterx-globals.h"
+#include "filterx/object-extractor.h"
+#include "filterx/object-json.h"
+#include "filterx/object-message-value.h"
+#include "filterx/object-null.h"
+#include "filterx/filterx-object.h"
+#include "filterx/object-dict-interface.h"
+#include "filterx/object-list-interface.h"
+#include "filterx/object-string.h"
+
+GQuark
+event_format_parser_error_quark(void)
+{
+  return g_quark_from_static_string("event-parser-error-quark");
+}
+
+Field
+field(FilterXFunctionEventFormatParser *self, int index)
+{
+  g_assert(index >= 0 && index < self->config.num_fields);
+  return self->config.fields[index];
+}
+
+static FilterXObject *
+parse_default(FilterXFunctionEventFormatParser *self, const gchar *value, GError **error, gpointer user_data)
+{
+  return filterx_string_new(value, -1);
+}
+
+FilterXObject *
+parse_version(FilterXFunctionEventFormatParser *self, const gchar *value, GError **error, gpointer user_data)
+{
+  const gchar *log_signature = signature_for_logtype(self->config.log_type);
+  gchar *colon_pos = strchr(value, ':');
+  if (!colon_pos || colon_pos == value)
+    {
+      g_set_error(error, EVENT_FORMAT_PARSER_ERROR, EVENT_FORMAT_PARSER_ERR_NO_LOG_SIGN,
+                  EVENT_FORMAT_PARSER_ERR_NO_LOG_SIGN_MSG, log_signature);
+      return FALSE;
+    }
+  if (!(strncmp(value, log_signature, colon_pos - value) == 0))
+    {
+      g_set_error(error, EVENT_FORMAT_PARSER_ERROR, EVENT_FORMAT_PARSER_ERR_LOG_SIGN_DIFFERS,
+                  EVENT_FORMAT_PARSER_ERR_LOG_SIGN_DIFFERS_MSG, value, log_signature);
+      return FALSE;
+    }
+  return filterx_string_new(++colon_pos, -1);
+}
+
+gboolean
+_set_dict_value(FilterXObject *out,
+                const gchar *key, gsize key_len,
+                const gchar *value, gsize value_len)
+{
+  FilterXObject *dict_key = filterx_string_new(key, key_len);
+  FilterXObject *dict_val = filterx_string_new(value, value_len);
+
+  gboolean ok = filterx_object_set_subscript(out, dict_key, &dict_val);
+
+  filterx_object_unref(dict_key);
+  filterx_object_unref(dict_val);
+  return ok;
+}
+
+gboolean unescape_value_separators(KVScanner *self)
```

Comment:
[optional]

This could be private. Also please move the function name to a new line for better `grep`-ability.